src/pool/srv_pool.c

```diff
@@ -4081,9 +4081,13 @@ static int pool_upgrade_props(struct rdb_tx *tx, struct pool_svc *svc,
 		if (rc)
 			D_GOTO(out_free, rc);
 		rc = pool_prop_read(tx, svc, DAOS_PO_QUERY_PROP_ALL, &prop);
-		if (rc)
+		if (rc) {
+			/* Prop might be allocated and returned even if rc != 0 */
```

Comment:
I guess it makes some sense, because it might read some parts of property and failed some,  fix pool_prop_read to free memory when error happen Looks better .

Reply:
I can look into this, and changing all the callers correspondingly, but it'll probably be a more involved change. Should it be done now as part of this coverity fix or in a separate patch (that I am happy to try to undertake). I can see an argument for either approach but perhaps it would be better to split it into 2 changes?

Reply:
That is fine for me.

Reply:
Personally I would prefer it to be part of the coverity patch, as it's an alternate way to solve the same problem. If pool_prop_read always frees the prop on exit, the cleanups in every caller aren't necessary anymore.